### PR TITLE
Add label for uploads to anaconda in Jenkins Pipeline

### PR DESCRIPTION
**Pull request by Pasarus**
**Description of work.**
Allows us to ensure files are labelled nightly from the nightly pipeline or main from the release pipeline based on env variables

**To test:**
Code-review

*There is no associated issue.*
*This does not require release notes* because **Developer Facing**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: buildconfig/Jenkins/Conda/publish-to-anaconda [hunk 1]
```diff
@@ -39,9 +40,9 @@ conda activate $CONDA_ENV_NAME
 for package in "$@"
 do
     # Skip the Workspace, Token, and Channel args, then upload every other package to the given User
-    if [ $package != $1 ] && [ $package != $2 ] && [ $package != $3 ] && [[ $package == *"mantid"* ]]
+    if [ $package != $1 ] && [ $package != $2 ] && [ $package != $3 ] && [ $package != $4 ] && [[ $package == *"mantid"* ]]
```
Comment (DavidFair) [inline buildconfig/Jenkins/Conda/publish-to-anaconda:43]: I'm sure there must be a better way of doing this, but can't think what at the moment...

**Review by DavidFair — DISMISSED** [review]

File: buildconfig/Jenkins/Conda/publish-to-anaconda (hunk 1 above)
Comment (martyngigg) [reply buildconfig/Jenkins/Conda/publish-to-anaconda:43]: Would it make sense to capture the first 4 arguments at the top alongside `WORKSPACE` and then `shift 4` to swallow those arguments leaving the packages?

Comment (Pasarus) [reply buildconfig/Jenkins/Conda/publish-to-anaconda:43]: $@ captures things that have been "shifted" past, so it wouldn't work here.

Comment (martyngigg) [reply buildconfig/Jenkins/Conda/publish-to-anaconda:43]: Oh. That's not my experience:

```
$ cat test.sh
echo $1
echo $2
echo $3


shift 2
echo

echo $@

$ bash test.sh 1 2 3
1
2
3

3
```

**Review by martyngigg — APPROVED** [review]
As agreed this morning adding labels is useful and great they can be added with minimal effort :)

**Review by DavidFair — APPROVED** [review]
Failure unrelated

**Pull request merged**